hal/common/critical.c

```diff
@@ -38,7 +34,7 @@ bool core_util_are_interrupts_enabled(void)
 #endif
 }
 
-void core_util_critical_section_enter()
+void core_util_critical_section_enter(void)
```

Comment:
please don't mix unrelated changes. You should submit them in separate pull requests. This is not a matter of convenience.